### PR TITLE
Add poster to restaurant object and deleting restaurant frontend

### PR DESCRIPTION
**Pull request by jyx11011**

### CODE REVIEW

File: src/main/java/com/google/step/datamanager/RestaurantManagerDatastore.java [hunk 1]
```diff
@@ -124,9 +127,10 @@ public void deleteRestaurant(long id) {
   private Restaurant transformEntityToRestaurant(Entity restaurantEntity) {
     String name = (String) restaurantEntity.getProperty("name");
     String photoUrl = (String) restaurantEntity.getProperty("photoUrl");
+    long posterId = (long) restaurantEntity.getProperty("posterId");
```
Comment (limli) [inline src/main/java/com/google/step/datamanager/RestaurantManagerDatastore.java:133]: if posterId doesn't exist (like past restaurant entities already in database), will this cause any errors? If yes, then might be better to maybe assign posterId to be -1, or something else, like the admin's Id

Comment (jyx11011) [reply src/main/java/com/google/step/datamanager/RestaurantManagerDatastore.java:133]: I add a admin user using a service account email in the restaurant generator.

Comment (limli) [reply src/main/java/com/google/step/datamanager/RestaurantManagerDatastore.java:133]: yea i mean currently there are some restaurants in the live datastore that doesn't have a posterId. Might need to remember to nuke everything after you merge, or manually add in posterIds to the ones without posterIds.

Comment (jyx11011) [reply src/main/java/com/google/step/datamanager/RestaurantManagerDatastore.java:133]: ok I see

**Review by limli — DISMISSED** [review]

**Review by SuyashLakhotia — DISMISSED** [review]

**Comment by SuyashLakhotia** [comment]
This is a really large PR. Due to time constraints, I'm approving it but please don't do this since it makes it **really** hard to parse and review in a meaningful way.

**Review by SuyashLakhotia — APPROVED** [review]
